elastic/kibana
# [ResponseOps] Flapping recovered alerts do not show up as active in the alerts table

Resolves https://github.com/elastic/kibana/issues/183735

## Summary

This bug is caused by event log docs not being written for "pending recovered" alerts on rules with "notify on every run" and "throttle" or notify when set to null. In this PR I remove the logic causing this in `xpack/plugins/alerting/server/lib/get_alerts_for_notification.ts` so now these alerts will be logged in the event log and show up in the ui. 

### Checklist



### To verify

- Create an ES Query rule without actions, or with `notifyWhen` set to "notify on every run" or "throttle". Get it to start flapping.
- Let the flapping alert start to recover and verify it's still reported as active in the UI until it's been recovered for X executions (the default is 4 if you don't mess with your flapping settings).

Ex. of what to expect
<img width="1461" alt="Screen Shot 2024-05-29 at 10 53 57 AM" src="https://github.com/elastic/kibana/assets/109488926/144dc22b-a1c7-460f-b430-a6d49771c9bd">
